Remove & Replace buttons

jryanlohmolder opened the pull request:
The "REMOVE" and "REPLACE" buttons are now working for the Tiled Browser. The data channels that are plotted in the scan window can be unchecked in the data channel table and new selections can be checked, with the changes being reflected in the scan window once the replace button has been clicked.

The _on_remove_clicked function was added to QTiledWidget. In addition the 'LabelNames' key was added to both the _on_add_clicked and _on_remove_clicked function.


_on_replace_clicked function was added and connect_self_signals was modified in QTiledWidget.py. _replaceSelection was modified in ScanWindow.py

jryanlohmolder committed:
REMOVE button now removes all selected scans from plot

jryanlohmolder committed:
REPLACE button is now working. Also cleaned up print statements.

jryanlohmolder committed:
Added * unpacking syntax back to QTiledWidget and QSourceSelector.

padraic-shafer commented on PyMca5/PyMcaGui/io/QTiledWidget.py:
```diff
@@ -464,7 +464,9 @@ def _on_add_clicked(self, *, emit=True):
                                   'm': channel_sel['m'],
-                                  'Channel List': channel_sel['Data Channel List']},
+                                  'Channel List': channel_sel['Data Channel List'],
+                                  'LabelNames': channel_sel['Data Channel List'],
```
IIRC, I think you'll need a copy since this list gets mutated in the ScanWindow(?).

```suggestion
                                  'LabelNames': channel_sel['Data Channel List'].copy(),
```

padraic-shafer replied:
Then, again it's probably better to do this in one place, like we did in an earlier iteration within the `TiledDataSource.get_data_object()` method.

https://github.com/padraic-shafer/pymca/blob/6492405ea9b5839271476cb3fb3ab130b368c36d/PyMca5/PyMcaCore/QTiledDataSource.py#L68-L73

padraic-shafer commented on PyMca5/PyMcaGui/io/QTiledWidget.py:
```diff
@@ -505,6 +605,7 @@ def on_table_changed(node_path_parts: Tuple[str]):
 
         self.model.url_changed.connect(self.model.on_url_changed)
 
+
```
```suggestion
```

AbbyGi commented on PyMca5/PyMcaCore/TiledDataSource.py:
```diff
@@ -147,6 +147,8 @@ def _ensure_max_dims(
         pass
 
+    def get_current_selections(self):
+        pass
```
Is this function used anywhere/needed?

AbbyGi reviewed (commented):
Looks pretty good. One small thing I noticed besides Padraic's comments.

jryanlohmolder committed:
Removed unnecessary function

Removed get_current_selections functions

jryanlohmolder committed:
Update QTiledWidget.py

changed 'Label Names': channel_sel['Data Channel List'] to 'Label Names': channel_sel['Data Channel List'].copy

jryanlohmolder committed:
Update QTiledWidget.py

Changed label names in both _on_remove_clicked and _on_replaced_clicked

AbbyGi reviewed (approved):
This looks good to me.
